open-reac/src/main/java/com/powsybl/openreac/parameters/input/VoltageLevelLimitsOverrideInput.java

```diff
@@ -32,21 +36,60 @@ public class VoltageLevelLimitsOverrideInput implements AmplInputFile {
 
     private final Map<String, Pair<Double, Double>> normalizedVoltageLimitsOverride;
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(VoltageLevelLimitsOverrideInput.class);
     private static final String OVERRIDE_ON_VOLTAGE_LEVEL = "Override on voltage level ";
+    private static final double VOLTAGE_LIMIT_LOW_THRESHOLD = 0.85;
+    private static final double VOLTAGE_LIMIT_HIGH_THRESHOLD = 1.15;
+    private static final double VOLTAGE_LIMIT_TOLERANCE = 5;
+    private static final DecimalFormat LIMIT_VALUE_FORMAT = new DecimalFormat("0.0", DecimalFormatSymbols.getInstance(Locale.ROOT));
 
-    public VoltageLevelLimitsOverrideInput(List<VoltageLimitOverride> voltageLimitsOverrides, Network network) {
+    public VoltageLevelLimitsOverrideInput(List<VoltageLimitOverride> voltageLimitsOverrides, Network network, ReportNode reportNode) {
         Objects.requireNonNull(voltageLimitsOverrides);
         Objects.requireNonNull(network);
         this.normalizedVoltageLimitsOverride = new HashMap<>();
-        transformToNormalizedVoltage(voltageLimitsOverrides, network);
+        transformToNormalizedVoltage(voltageLimitsOverrides, network, reportNode);
+    }
+
+    private void checkLimitsInNominalVoltageRange(Network network, ReportNode reportNode) {
+        // check that the limits are in the nominal voltage range [0.85 * nominal_V - 5, 1.15 * nominal_V + 5]
+        Map<String, VoltageLevelLimitsInfos> voltageLevelsWithLimitsOutOfNominalVRange = new HashMap<>();
+        for (Map.Entry<String, Pair<Double, Double>> entry : normalizedVoltageLimitsOverride.entrySet()) {
+            String voltageLevelId = entry.getKey();
+            VoltageLevel voltageLevel = network.getVoltageLevel(voltageLevelId);
+            double nominalV = voltageLevel.getNominalV();
+            double lowLimit = entry.getValue().getFirst() * nominalV;
+            double highLimit = entry.getValue().getSecond() * nominalV;
+
+            if (lowLimit < VOLTAGE_LIMIT_LOW_THRESHOLD * nominalV - VOLTAGE_LIMIT_TOLERANCE ||
+                lowLimit > VOLTAGE_LIMIT_HIGH_THRESHOLD * nominalV + VOLTAGE_LIMIT_TOLERANCE ||
+                highLimit < VOLTAGE_LIMIT_LOW_THRESHOLD * nominalV - VOLTAGE_LIMIT_TOLERANCE ||
+                highLimit > VOLTAGE_LIMIT_HIGH_THRESHOLD * nominalV + VOLTAGE_LIMIT_TOLERANCE) {
+                voltageLevelsWithLimitsOutOfNominalVRange.put(voltageLevelId, new VoltageLevelLimitsInfos(voltageLevelId, lowLimit, highLimit, nominalV));
+            }
+        }
+
+        if (!voltageLevelsWithLimitsOutOfNominalVRange.isEmpty()) {
+            reportNode.newReportNode()
```

Comment:
I recently realized that there was a Reports class where you should do the `reportNode.newReportNode(). ...` type of operations. Could you move your report node creations to this class?

Reply:
Done